pkg/core/serving/knative/servingrun.go

```diff
@@ -326,9 +382,15 @@ func (r *servingRun) createService(s *openfunction.Serving, cm map[string]string
 	return &service
 }
 
-func getFunctionName(s *openfunction.Serving) string {
-
-	return s.Labels[constants.FunctionLabel]
+func (r *servingRun) needCreateProxy(s *openfunction.Serving) bool {
+	mode := common.GetDaprSidecarInjectMode(s)
+	if mode != common.PerFuncMode {
+		return false
+	}
+	if s.Spec.Outputs != nil || s.Spec.Annotations[common.DaprEnabled] == "true" {
```

Comment:
- Async Function might only have `inputs`, the `outputs` is optional https://github.com/OpenFunction/samples/blob/main/functions/async/bindings/kafka-input/kafka-input.yaml#L48
- Synch function's `outputs` is optional and it doesn't have `inputs`

Reply:
only use for sync function. aysnc function will create dapr proxy if mode is PerFuncMode.

Reply:
we need a proxy when:
- The function's injection mode is `per function` and 
- it has `inputs` or `outputs` defined

It's not relevant to whether the `DaprEnabled` annotation is true?

Reply:
to support some non-standard usage, e.g: wasm with rust.